### PR TITLE
fix(client): changed the reset button in JS course

### LINKED ISSUE #48986: JS course button "Reset All Code" should probably be called "Reset This Lesson" instead
### Describe the Issue

In the JS course (such as the Basic Javascript course), the button to "Reset All Code" leads to a dialog titled "Reset This Lesson" which contains a button titled "Reset this lesson". 
I believe that "Reset this lesson" is a clearer and truer description of what clicking on "Reset All Code" is meant to do so I opened this issue to discuss that. I've also heard feedback before on the forum where someone was afraid to press on "Reset All Code" button because they thought it would delete all their progress.
For this reason and for the reason for having a consistent and predictable user-interface, I suggest the button "Reset All Code" be renamed to match the dialog "Reset This Lesson"

### Affected Page

all js course pages

### Steps to Reproduce

click on the button in any of the JS course steps to see the dialog 

### Expected behavior

please see the description of the issue

### Screenshots

js-reset-button

### System

laptop devices and other large screens

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rahulsiloniya**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #48986

The button label showed "Reset all code" which was misleading for some users. It is now changed to "Reset this lesson".

### CODE REVIEW

File: client/i18n/locales/english/translations.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
     "ask-for-help": "Ask for Help",
     "create-post": "Create a help post on the forum",
     "cancel": "Cancel",
-    "reset-lesson": "Reset this lesson",
+    "reset-lesson": "Reset this step",
```
Comment (Sboonny) [inline client/i18n/locales/english/translations.json:50]:
```suggestion
    "reset-lesson": "Reset this lesson",
```

File: client/src/templates/Challenges/classic/lower-jaw.tsx [hunk 2]
```diff
@@ -190,8 +190,8 @@ const LowerJaw = ({
         <div className='lower-jaw-icon-bar'>
           <button
             className='btn fade-in'
-            title={t('buttons.reset-code')}
-            aria-label={t('buttons.reset-code')}
+            title={t('buttons.reset-lesson')}
+            aria-label={t('buttons.reset-lesson')}
```
Comment (Sboonny) [inline client/src/templates/Challenges/classic/lower-jaw.tsx:194]:
```suggestion
            title={t('buttons.reset-step')}
            aria-label={t('buttons.reset-step')}
```

**Review by Sboonny — COMMENTED** [review]
in translation.json file, you need to change `"reset-code": "Reset All Code",` to `"reset-step": "Reset the step"`, in line 56.

**Review by Sboonny — APPROVED** [review]
#### :sparkles: :tada: **AWESOME!** :tada: :sparkles:

 Hi @rahulsiloniya, thanks for this pull request and for contributing to the code-base for the first time. We are looking forward to more contributions from you in the future.

 Cheers & happy contributing!

File: client/i18n/locales/english/translations.json [hunk 3]
```diff
@@ -53,7 +53,7 @@
     "check-code": "Check Your Code (Ctrl + Enter)",
     "check-code-2": "Check Your Code",
     "reset": "Reset",
-    "reset-code": "Reset All Code",
+    "reset-step": "Reset this step",
```
Comment (Sembauke) [inline client/i18n/locales/english/translations.json:56]:
```suggestion
    "reset-step": "Reset This Step",
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

File: client/i18n/locales/english/translations.json (hunk 3 above)
Comment (rahulsiloniya) [reply client/i18n/locales/english/translations.json]: Completed the above suggested changes to branch.

**Review by Sembauke — APPROVED** [review]

**Pull request merged**